### PR TITLE
Validation additions

### PR DESCRIPTION
**Pull request by pieterbos**
Two missing validations:

- a use_node cannot point to another use_node, and certainly not to itself
- the type of a use_node was not validated, so you could add a POINT_EVENT to CLUSTER.items without a validation error, but only if the POINT_EVENT was a use node

Added failing tests and new validations.

### CODE REVIEW

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java [hunk 1]
```diff
@@ -1,31 +1,48 @@
...
             addMessageWithPath(ErrorType.VUNP, cObject.path(), I18n.t("Use_node (C_COMPLEX_OBJECT_PROXY) points to a path that cannot be found: {0}", cObject.getTargetPath()));
+        } else if (replacements.size() > 1) {
+            addMessageWithPath(ErrorType.VUNP, cObject.path(), I18n.t("Use_node (C_COMPLEX_OBJECT_PROXY) points to a path that resolves to more than one object"));
         } else {
-            CComplexObject replacement = complexObjectProxyReplacement.getReplacement();
-            if(!combinedModels.rmTypesConformant(replacement.getRmTypeName(), cObject.getRmTypeName())) {
-                addMessageWithPath(ErrorType.VUNT, cObject.path(), I18n.t("Use_node (C_COMPLEX_OBJECT_PROXY) points to type {0}, which does not conform to type {1}", replacement.getRmTypeName(), cObject.getRmTypeName()));
+            ArchetypeModelObject replacement = replacements.get(0);
+            if(replacement instanceof CComplexObjectProxy) {
+                addMessageWithPath(ErrorType.VUNP, cObject.path(), I18n.t("Use_node (C_COMPLEX_OBJECT_PROXY) points to a use_node (C_COMPLEX_OBJECT_PROXY). That is not allowed"));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java]: This is already covered by the else clause below. Maybe remove it?

File: tools/src/test/resources/adl2-tests/validity/rm_checking/openEHR-EHR-OBSERVATION.VCORMT_rm_non_conforming_type_use_node.v1.0.0.adls [hunk 2]
```diff
@@ -0,0 +1,71 @@
+archetype (adl_version=2.0.5; rm_release=1.0.4)
```
Comment (J3173) [inline tools/src/test/resources/adl2-tests/validity/rm_checking/openEHR-EHR-OBSERVATION.VCORMT_rm_non_conforming_type_use_node.v1.0.0.adls]: Is this archetype used anywhere?

File: tools/src/test/resources/adl2-tests/validity/structure/openEHR-TEST_PKG-EVALUATION.VUNP_use_node_points_to_use_node.v1.0.0.adls [hunk 3]
```diff
@@ -0,0 +1,70 @@
+archetype (adl_version=2.0.7; rm_release=1.0.2)
```
Comment (J3173) [inline tools/src/test/resources/adl2-tests/validity/structure/openEHR-TEST_PKG-EVALUATION.VUNP_use_node_points_to_use_node.v1.0.0.adls]: Is this archetype used anywhere?

Comment (pieterbos) [reply tools/src/test/resources/adl2-tests/validity/structure/openEHR-TEST_PKG-EVALUATION.VUNP_use_node_points_to_use_node.v1.0.0.adls]: Everything in the validity directory is automatically used by the test. If it contains regression = <"SOME_ERROR">, the test asserts that that error is present. If "PASS", it must validate. If you remove the code changes, you will see this archetype causing a failed test.

File: tools/src/test/resources/adl2-tests/validity/rm_checking/openEHR-EHR-OBSERVATION.VCORMT_rm_non_conforming_type_use_node.v1.0.0.adls (hunk 2 above)
Comment (pieterbos) [reply tools/src/test/resources/adl2-tests/validity/rm_checking/openEHR-EHR-OBSERVATION.VCORMT_rm_non_conforming_type_use_node.v1.0.0.adls]: Everything in the validity directory is automatically used by the test. If it contains regression = <"SOME_ERROR">, the test asserts that that error is present. If "PASS", it must validate. If you remove the code changes, you will see this archetype causing a failed test.

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java (hunk 1 above)
Comment (pieterbos) [reply tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java]: Better like this? :)

File: tools/src/test/resources/adl2-tests/validity/rm_checking/openEHR-EHR-OBSERVATION.VCORMT_rm_non_conforming_type_use_node.v1.0.0.adls (hunk 2 above)
Comment (pieterbos) [reply tools/src/test/resources/adl2-tests/validity/rm_checking/openEHR-EHR-OBSERVATION.VCORMT_rm_non_conforming_type_use_node.v1.0.0.adls]: 'the test' being com.nedap.archie.archetypevalidator.BigArchetypeValidatorTest :)

Comment (J3173) [reply tools/src/test/resources/adl2-tests/validity/rm_checking/openEHR-EHR-OBSERVATION.VCORMT_rm_non_conforming_type_use_node.v1.0.0.adls]: Ah, I didn't know that. Nice method.

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java (hunk 1 above)
Comment (J3173) [reply tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java]: Yes 👍

**Pull request merged**